drivers/stepper/gpio_stepper/gpio_stepper_controller.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_stepper
+
+#include "zephyr/drivers/gpio.h"
+#include <zephyr/kernel.h>
+#include "zephyr/drivers/stepper.h"
+#include <zephyr/logging/log.h>
+#include "stdlib.h"
+
+LOG_MODULE_REGISTER(gpio_stepper_motor_controller, CONFIG_STEPPER_LOG_LEVEL);
+
+struct gpio_stepper_motor_controller_data {
+	struct k_work_delayable stepper_dwork;
+	const struct gpio_dt_spec *gpio_in_pins;
```

Comment:
The name `in_pins` seems antithetical to the pin configuration, as these are not configured as input pins, they are outputs to the stepper motor controller. It is a question of perspective, but using a direction agnostic name like `phase_pins` or `control_pins` removes such confusion